src/tunnel.rs

```diff
@@ -205,32 +223,25 @@ macro_rules! make_tunnel_type {
             #[napi]
             pub async fn close(&self) -> Result<()> {
                 debug!("{} closing, id: {}", stringify!($wrapper), self.id);
-                // drop our internal reference to the tunnel, giving drop control of the tunnel back to the nodejs.
-                GLOBAL_TUNNELS.lock().await.remove(&self.id);
 
                 // we may not be able to lock our reference to the tunnel due to the forward_* calls which
                 // continuously accept-loop while the tunnel is active, so calling close on the Session.
-                self.session.close_tunnel(self.id.clone())
+                let res = self.session.close_tunnel(self.id.clone())
                     .await
-                    .map_err(|e| napi_err(format!("error closing tunnel: {e:?}")))
+                    .map_err(|e| napi_err(format!("error closing tunnel: {e:?}")));
+
+                // drop our internal reference to the tunnel after awaiting close
+                GLOBAL_TUNNELS.lock().await.remove(&self.id);
```

Comment:
This feels like something that should happen in a `Drop` implementation, not `close`. Or maybe in `ObjectFinalize`? But it might not really matter all that much, because what else can you really do with an already-closed tunnel anyway?

Reply:
That was basically the default behavior, but ran into the issue of NodeJS losing all references to this wrapper object since the http server strips out the native socket and drops everything around what was passed in, and then NodeJS would GC the wrapper object and the ngrok-rs tunnel inside would die.

So with this the tunnels will stay up in that situation without the user having to keep around dummy references, and the user still has the power to drop if they do keep around a reference and close it (which the trailer's `listen` does by default on SIGINT).